### DISCUSSION TITLE
Customize views based on entity type?

### DISCUSSION THREAD

**Discussion by saumier** [zazuko/trifid#86] [Q&A] [answered]
I was wondering how you would approach customizing views in Trifid based on the Entity type.  For example, if a URI is a Person entity, the view should render with the photo of the person, and if a Organization entity is requested, the view should display a different layout with a map. We are using opaque URIs so there is no "path" that indicates the type of requested entity before hand. 

The ability to customize views based on entity type seems like a common pattern so I am sure there must be a way to do this elegantly with Trifid ;-).

**Comment by bergos** [comment, marked as answer]
@saumier The main work to render the triples in the html page is done by the [template](https://github.com/zazuko/trifid/blob/master/views/header.html) that injects the triples as JSON-LD into a `script` element and the [render-ld.js](https://github.com/zazuko/trifid-renderer-simple/blob/master/public/js/render-ld.js) script, which creates the html. It would be possible to replace the standard renderer, if you want to replace the standard output with custom code. It's also possible to just add more js code for additional output. Maybe start with the last option, once you have something useful running, you can decide if you want to replace the default rendering. [Here](https://github.com/geoadmin/linkeddata-trifid/blob/develop/public/js/map.js) is an example js code that adds a map if there is a triples with the main subject and the predicate `http://schema.org/hasMap`. The result looks like [this](https://ld.geo.admin.ch/boundaries/municipality/1).

**Reply by saumier** [reply]
@bergos Thanks.  In your message, the link to the example js code is not working. Probably because geoadmin's GitHub repository 'linkeddata-trifid' is not public. Is it possible to share the example js code?

**Reply by ktk** [reply]
@saumier I copied it to a gist here https://gist.github.com/ktk/1289a05f06a321fdd7619b0a678b43af
